.github/workflows/sphinx.yml

```diff
@@ -35,16 +35,26 @@ jobs:
         run: |
           pip install -r requirements.txt
 
+      # Action to download artifacts from a different workflow (analysis.yml)
+      - name: Download figures artifact
```

Comment:
This introduces a dependency between workflows that we now need to manage. Is it not possible to upload documentation into one folder on the published page from one workflow and figures from another? Or does it purge the page when a new thing is uploaded?

If we do need to merge the results from multiple workflows to publish, then that publish step should probably be a separate workflow triggered when one or both of the source workflows complete. I looks like that's possible with [this](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_run) and it even allows passing artifacts around, so perhaps that's how it's intended?

Another option would be to add this doc building to the analysis workflow. I'm not too happy with that because it seems like it's two separate things, but maybe that's the publishing workflow?

Reply:
> Is it not possible to upload documentation into one folder on the published page from one workflow and figures from another?

Not sure if I understand this part, can you elaborate more? The `sphinx.yml` workflow downloads the figures from the last completed `analysis` workflow. And then publish the website. So this doesn't require the current `analysis` job to be completed, it captures the figures from the last successful one.

> I looks like that's possible with [this](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_run) and it even allows passing artifacts around, so perhaps that's how it's intended?

This will be very useful to keep the website updated with the current figures produced by `analysis.yml`.
I'm thinking of deploying the website twice: a quick one that takes ~50 seconds and uses the last available figures artifact. And the second one is when the new figures are uploaded. What do you think?
```
on:
  push:
    branches:
      - 'main'
  workflow_run:
    workflows: ["analysis.yml"]
    types:
      - completed
```

Reply:
This is really good. It would be nice if the upload were a separate workflow, but maybe that's a followup improvement.

What happens in the case of a build failure on either branch?

I think the last thing is it would be nice to use the official github [method](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#using-data-from-the-triggering-workflow) of accessing data from the other workflows. It's much more verbose but is the official method. Let me know your thoughts.

Reply:
> This is really good. It would be nice if the upload were a separate workflow, but maybe that's a followup improvement.

I did the build and deployment on the same workflow because each step depends on the previous one. Not sure how to further improve this, maybe upload the build folder as an artifact and use it on a separate deployment workflow?


>  What happens in the case of a build failure on either branch?

If the build fails for any reason the website will be available with the last successful build.

> I think the last thing is it would be nice to use the official github [method](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#using-data-from-the-triggering-workflow) of accessing data from the other workflows. It's much more verbose but is the official method. Let me know your thoughts.

Done!

Reply:
Fixed! Now if the analysis fails, the figures step will be skipped and the generation will work.